pkg/operator/controller/callbacks.go

```diff
@@ -132,6 +107,28 @@ func reconcileDeleteControllerDeployment(args *callbacks.ReconcileCallbackArgs)
 	return nil
 }
 
+func deleteCRDs(args *callbacks.ReconcileCallbackArgs) error {
+	crdNames := []string{
+		"dataimportcrons.cdi.kubevirt.io",
+		"datavolumes.cdi.kubevirt.io",
+		"objecttransfers.cdi.kubevirt.io",
+	}
+	crdsExist := false
+	for _, crdName := range crdNames {
+		crd := &extv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: crdName}}
```

Comment:
Maybe changing this to the following to address unit test issues?

Get CRD if not exist continue
If CRD.deletionTimestamp == nil{
  Delete CRD
  Get CRD if not exist continue
}
crdsExist = true

Reply:
So the issue was caused by repeated Delete() of CRD which already had DeletionTimestamp? Late night games...
Fixed